tests/test_pds4_create_xml_index.py

```diff
@@ -0,0 +1,460 @@
+                '--elements-file',
+                str(root_dir / 'samples/sample_elements.txt'),  
+                '--output-file',     
+             ]
+            ),
+            # Okay, basic capture
+            (str(test_files_dir / 'elements_file_success_2.csv'),
+             'elements_file_2.csv',
+             [
+                str(test_files_dir),
+                'tester_label_2.xml',
+                '--elements-file',
+                str(root_dir / 'samples/element_2.txt'),  
+                '--output-file',     
+             ]
+            ),
+            # Multiple partial captures
+            (str(test_files_dir / 'elements_file_success_3.csv'),
+             'elements_file_3.csv',
+             [
+                str(test_files_dir),
+                'tester_label_2.xml',
+                'tester_label_3.xml',
+                '--elements-file',
+                str(root_dir / 'samples/element_3.txt'),  
+                '--output-file',     
+             ]
+            ),
+
+            # exclusion test
+            (str(test_files_dir / 'elements_file_success_4.csv'),
+            'elements_file_4.csv',
+             [
+                str(test_files_dir),
+                'tester_label_*.xml',
+                '--elements-file',
+                str(root_dir / 'samples/element_4.txt'),  
+                '--output-file',     
+             ]
+            )
+
+
+        ]
+                        )
+                    
+def test_elements_file(golden_file, new_file, cmd_line):
+    # Create a temporary directory in the same location as the test_files directory
+    with tempfile.TemporaryDirectory(dir=test_files_dir.parent) as temp_dir:
+        temp_dir_path = Path(temp_dir)
+        
+        # THE PATH TO THE NEW FILE
+        path_to_file = temp_dir_path / new_file
+        # Call main() function with the simulated command line arguments
+        cmd_line.append(str(path_to_file))
+        tools.main(cmd_line)
+
+        # Assert that the file now exists
+        assert os.path.isfile(path_to_file)
+
+        # Open and compare the two files
+        with open(path_to_file, 'rb') as created:
+            formed = created.read()
+
+        with open(golden_file, 'rb') as new:
+            expected = new.read()
+
+        assert formed == expected
+
+
+
+# Set parameters values that you would like to pass into test_simplify_xpaths.
+# In this case, we are running the same test with different sets of
+# golden_file, new_file, and cmd_line.
+@pytest.mark.parametrize(
+        'golden_file,new_file,cmd_line',
+        [
+            (str(test_files_dir / 'simplify_xpaths_success_1.csv'),
+             'simplify_xpaths_1.csv',
+             [
+                str(test_files_dir),
+                'tester_label_1.xml',
+                '--simplify-xpaths',
+                '--output-file',     
+             ]
+            ),
+
+            (str(test_files_dir / 'simplify_xpaths_success_2.csv'),
```

Comment:
I agree that using `--dump-available elements` in certain unit tests is a better solution. This will be applied to the unit tests that can benefit from this feature (especially `--simplify-xpaths`)